P8/DataConverters/ShapeletHistogramConverter/FeatureExtraction/src/types/Series.h

```diff
@@ -0,0 +1,18 @@
+#ifndef FEATUREEXTRACTION_SERIES_H
+#define FEATUREEXTRACTION_SERIES_H
+
+#include <vector>
+#include <algorithm>
+
+struct Series : public std::vector<double> {
+    // Returns the minimum value found in the series
+    [[nodiscard]] inline double Min() const { return *std::min_element((*this).begin(), (*this).end()); }
+    [[nodiscard]] inline double Max() const { return *std::max_element((*this).begin(), (*this).end()); }
+    // Applies a function all values
+    template<typename T>
+    inline void Apply(T&& lambda) {
+        std::for_each((*this).begin(), (*this).end(), lambda);
+    }
```

Comment:
Now this seems like a rather risky method, im not 100p sure how you are using it, but it seems like a method that can be problematic.
(This is more a note and not a complaint tho)